### ISSUE TITLE
unmaintained version of tinyyaml (and timestamps)

### ISSUE THREAD

**Issue by TeXhackse** [kalekje/YAMLvars#1]
I just stumbled across this package by searching CTAN for tinyyaml because we did some adjustments for the markdown package. 

The used version of tinyyaml seems to be unmaintained. 
Therefore I suggest to switch to the maintained one https://github.com/api7/lua-tinyyaml which we already include as `markdown-tinyyaml.lua` within TeX Live. Maybe there's also a way to not ship 2 copies of the same library. 

This package is currently also converting iso-dates (yyyy-mm-dd) to timestamps (corresponding issue for markdown: https://github.com/Witiko/markdown/issues/114). 

I just added an option to disable that feature within the mentioned tinyyaml repo https://github.com/api7/lua-tinyyaml/pull/21 which might also be useful for this package.

**Comment by kalekje** [comment]
Thank you for sharing this, much appreciated. The version I have is a copy from the repo you linked, I believe.

Since the markdown package already uses the tinyyaml package, I am willing to adjust my package and omit tinyyaml.lua to avoid redundancy, and have YAMLvars depend on markdown. I assume the markdown package will have a wider userbase so it's only fair.

**Comment by TeXhackse** [comment]
Thanks for the fast reply. 

Hm, I am wondering if it also might be an option to ship the library separate. If not you might want to add an error message pointing this dependency. Users might otherwise be confused since TeX does not have a dependency management and a lot of users don't install a full distribution. 

In case there is anything more I can do about this, let me know. My intention was mostly to inform you about this so we can see what's the best solution for all.  :)

**Comment by kalekje** [comment]
You are correct that it would be best to put an error message. I guess the brute force way would be `\RequirePackage{markdown}` in my `.sty` file vs. `require'markdown-tinyyaml.lua'` in my `.lua` file, at least those with auto-install would not have to worry. That being said,

I have no problem if tinyyaml is shipped as a separate package. Since you folks are maintaining it, if you choose that route I support it and would adjust YAMLvars accordingly.

**Comment by TeXhackse** [comment]
`RequirePackage` would load the complete package. Which would slow the compilation process down … only to test if the file exists. I'd simply check if `markdown-tinyyaml` exists and throw an error message otherwhise. There's a bunch of packages which do so. Users then can manually install the package. This is less convenient than the auto install but it's still less impact. 

We will discuss the separate thing and I'll return.

**Comment by kalekje** [comment]
That's fair, I was actually hoping to incorporate some `markdown` features into `YAMLvars` anyways though, so in the future it may be the path I take regardless. Just curious if you can share an example on how to check for a Lua package's availability? I suppose the error message is thrown with `tex.print('\\PackageError{......}')`?

**Comment by TeXhackse** [comment]
I'd check that in the beginning of the package so either directly after the `\ProvidesPackage`. Since you load a bunch of packages yourself it's helpful to throw the error as soon as possible.

You can simply check if it exists by
```
	\IfFileExists{markdown-tinyyaml.lua}{}{<ERROR>}
```
or in expl3
```
	\file_if_exist:nF {markdown-tinyyaml.lua}
	{\msg_error:nn {<module>} {<name>}}
```
I even think if this is not a case where one could throw a fatal error to abort the compilation process because most people use the nonstopmode anyway and this package won't work without the library.

**Comment by TeXhackse** [comment]
Oh and I should add. We discussed my thoughts on a separate package, and it's not useful as long as CTAN doesn't support „only lua library packages“ and only focuses on the TeX part about it. So you can either load markdown's lua-tinyyaml or ship and update your own. 

Renaming would only be possible if you rename yours first, so we don't have two files of the same name within CTAN at once, and by the unique name markdown-tinyyaml we don't get conflicting copies anyway.

**Comment by kalekje** [comment]
Ahh that way is much easier than doing it in lua, thanks for sharing.

I do agree with your sentiments about a standalone package.

On my next release of YAMLvars, I will no longer include `tinyyaml.lua` and will instead depend on `markdown-tinyyaml.lua`. I should confirm though: there are no changes in `markdown-tinyyaml.lua` that may affect my package, right? In otherwords, `markdown-tinyyaml.lua` is essentially the same as `tinyyaml.lua`?

**Comment by TeXhackse** [comment]
`markdown-tinyyaml.lua` contains exactly the content from tinyyaml.lua. All the markdown repo is doing to it is adding the License Info and renaming the file. The repo directly loads the linked repo for the tinyyaml package as a submodule so others know this. 

The renaming was caused by the fact that the author wanted to ensure to avoid conflicts. One could locally install additional libraries. A renaming would be possible after you don't ship the `tinyyaml.lua` anymore with your package but still users might have other libraries with the same name withing their `$LUAINPUTS`. So we discussed it and thought it's much easier to keep the name to `markdown-tinyyaml`. There might be other usecases.

Concerning the libraries content:
The update ncludes some bug fixes and the extension for the option to disable the timestamp datatype. You can have a look at the repo to see all of them (https://github.com/api7/lua-tinyyaml)

This can be enabled. The corresponding commig in markdown is https://github.com/Witiko/markdown/pull/116/commits/75f6f203fc03018e79d76a1eed5da55cb185697a

If you're not in a rush I could also prepare a PR during one of the next evenings.

**Comment by kalekje** [comment]
Great, I will update YAMLvars accordingly. Will be updated on github tonight hopefully and I'll upload to CTAN ASAP.

**Comment by TeXhackse** [comment]
Alright, if you need anything else, feel free to ask. :)
